src/components/organisms/EarthEditor/TagPane/SceneTagPane/hooks.ts

```diff
@@ -0,0 +1,39 @@
+import { useCallback } from "react";
+
+import useCommonHooks from "../commonHooks";
+
+export default () => {
+  const {
+    loading,
+    removeTagGroupFromScene,
+    removeTagItemFromScene,
+    createTagGroup,
+    createTagItem,
+    sceneTagGroups,
+    updateTagGroup,
+  } = useCommonHooks();
+
+  const handleCreateTagGroup = useCallback(
+    async (label: string) => {
+      createTagGroup(label);
+    },
+    [createTagGroup],
+  );
+
+  const handleCreateTagItem = useCallback(
+    async (label: string, tagGroupId: string) => {
+      await createTagItem(label, tagGroupId);
+    },
+    [createTagItem],
+  );
```

Comment:
**NOTE: You don't need to reply if you think this is okay. Just wanted to point this out just in case (一応)**

Both createTagGroup and createTagItem are async/await functions (coming from commonHooks). Not sure about degradation or slowdown nesting async/await useCallback functions, but either way if this is part of what you'll clean up later maybe you can mark this as resolved. (Better to just import commonHooks into the component directly, no?

Reply:
Yes, these parts are really messy and wanted to be refactored.
I'll open the refactoring PR next week.